tests/phpunit/tests/functions/wpNonceField.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * Tests for the wp_nonce_field() function.
+ *
+ * @since 6.0.0
+ *
+ * @group functions.php
+ * @covers ::wp_nonce_field
+ */
+class Tests_Functions_wpNonceField extends WP_UnitTestCase {
+
+	/**
+	 * @ticket 55578
+	 */
+	public function test_wp_nonce_field() {
+
+		wp_nonce_field();
+		$this->expectOutputRegex( '#^<input type="hidden" id="_wpnonce" name="_wpnonce" value=".{10}" /><input type="hidden" name="_wp_http_referer" value="" />$#' );
+	}
+
+	/**
+	 * @ticket 55578
+	 *
+	 * @dataProvider data_wp_nonce_field
+	 */
+	public function test_wp_nonce_field_return( $action, $name, $referer, $expected_reg_exp ) {
+
+		$this->assertMatchesRegularExpression( $expected_reg_exp, wp_nonce_field( $action, $name, $referer, false ) );
+	}
+
+	/**
+	 * @ticket 55578
+	 */
+	public function data_wp_nonce_field() {
+
+		return array(
+			'default'     => array(
+				- 1,
+				'_wpnonce',
+				true,
+				'#^<input type="hidden" id="_wpnonce" name="_wpnonce" value=".{10}" /><input type="hidden" name="_wp_http_referer" value="" />$#',
```

Comment:
```suggestion
				'action'           => - 1,
				'name'             => '_wpnonce',
				'referer'          => true,
				'expected_reg_exp' => '#^<input type="hidden" id="_wpnonce" name="_wpnonce" value=".{10}" /><input type="hidden" name="_wp_http_referer" value="" />$#',
```
Adding keys for each parameter in the datasets makes it easier for others to add datasets in the future.